### DISCUSSION TITLE
JS in custom layouts possible?

### DISCUSSION THREAD

**Discussion by jbmueller0584** [microsoft-search/pnp-modern-search#2144] [Q&A]
Hi dear Community,

is it possible to use JavaScript plain or any JS framework inside a [custom layout](https://microsoft-search.github.io/pnp-modern-search/scenarios/create-your-first-custom-template/)?

I'm asking because we try to build a flyout for every single search result for a page which uses the modern search webpart. We already implemented it with plain CSS. But is there a way to use CSS + JS for implementing such a flyout for a search result?

Thank you for your feedback!

Best regards,
Joerg

**Comment by wobba** [comment]
Not directly as that would not be secure. See https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/ for an alternative solution.
